aea/multiplexer.py

```diff
@@ -201,30 +218,36 @@ async def connect(self) -> None:
                 self.logger.debug("Multiplexer already connected.")
                 return
             try:
+                self.connection_status.set(ConnectionStates.connecting)
                 await self._connect_all()
-                enforce(self.is_connected, "At least one connection failed to connect!")
-                self._connection_status.is_connected = True
+
+                if all(c.is_connected for c in self._connections):
+                    self.connection_status.set(ConnectionStates.connected)
+                elif not all(
```

Comment:
why not simply `else`? If not all connections are connected then it means `connect` of at least one connection has been implemented wrongly or failed so we should raise?

Reply:
updated, but should discuss